pkg/cri/resource-manager/config/server.go

```diff
@@ -90,7 +123,8 @@ func (s *server) Stop() {
 // GetNode gets K8s node object.
 func (s *server) SetConfig(ctx context.Context, req *v1.SetConfigRequest) (*v1.SetConfigReply, error) {
 	s.Debug("REQUEST: %s", req)
-	return &v1.SetConfigReply{}, s.setConfigCb(&RawConfig{NodeName: req.NodeName, Data: req.Config})
+	s.requests <- req
+	return &v1.SetConfigReply{}, nil
```

Comment:
https://github.com/intel/cri-resource-manager/pull/53#issue-332274584https://github.com/intel/cri-resource-manager/pull/53#issue-332274584

Reply:
I can only quote myself from #53 , which is the PR introducing that change and explains the reasoning and potential remedy.
```
Also always reply with success for configuration updates. The old behavior of replying with an error and then indefinitely trying to resubmit the exact same already rejected data does not make sense. We can start reporting configuration failures asynchronously once we figure out how and where to propagate the error back. In the stock agent case, it could be propagated back to the offending ConfigMap.
```

Reply:
About getting stuck in the old configuration... IMO there is exactly zero point in trying to push a rejected configuration over and over to cri-resmgr, hoping that it suddenly would start magically working.

And that is what was happening with the old implementation. It did not differentiate between a failed configuration push and a successful configuration push that failed to get activated. It just kept retrying forever for both. It is the right thing to do for the former case and the wrong thing to do for the latter.

Reply:
That said, let me rework this so that I move the ratelimiting also to the agent side. That way we won't waste the reply error.

Reply:
See PR #53 (rate-limiting moved to the agent side) and #56 (separate errors for update delivery vs. application errors).